protocol/messenger_wallet.go

```diff
@@ -73,7 +77,30 @@ func (m *Messenger) watchWalletBalances() {
 	}()
 }
 
+func (m *Messenger) UpdateKeypairName(keyUID string, name string) error {
+	if keyUID == m.account.KeyUID && name != m.account.Name {
+		// profile keypair name must always follow profile display name
+		return ErrCannotChangeKeypairName
```

Comment:
ErrNameNotEqualAcountName sounds better? follow this way we don't need extra comment, ErrCannotChangeKeypairName is really not easy to understand for me

Reply:
Here is not a question whether is the name same or not, but more about requirement, so user cannot update profile keypair name directly, the only way to update it is by changing a display name. This error is descriptive enough and informs one who tries to use this `UpdateKeypairName` call to update a profile keypair that he "cannot change profile keypair name".

Reply:
May I express my thoughts here?

Generally the presence of any error communicates that an action failed. We make errors descriptive / have different types of error because we want to communicate why an action failed.

In this specific case, if we communicate why the action `UpdateKeypairName` failed the user of the function can make a correction to their use of the function. Otherwise we would just return an empty error, because we already know that the action failed due to the presence of an error.

The more details as to why something failed the more useful the error is at helping the user of the code to debug their issue.